tests/handlers/test_user_directory.py

```diff
@@ -639,6 +640,64 @@ def _add_user_to_room(
             self.hs.get_storage().persistence.persist_event(event, context)
         )
 
+    def test_making_room_public_doesnt_alter_directory_entry(self):
+        """Per-room names shouldn't go to the directory when the room becomes public.
```

Comment:
(feels a little bit funny testing that something *doesn't* happen :smile:, but I guess since it historically did, it makes sense)

Reply:
Possibly overkill. I'm just in the habit of reproducing bugs with a test case.

Reply:
I agree, though this one feels funny in a way I can't quite describe. :)